src/main/java/org/json/JSONParserConfiguration.java

```diff
@@ -4,23 +4,49 @@
  * Configuration object for the JSON parser. The configuration is immutable.
  */
 public class JSONParserConfiguration extends ParserConfiguration {
+    /**
+     * Used to indicate whether to overwrite duplicate key or not.
+     */
+    private boolean overwriteDuplicateKey;
 
-  /**
-   * Configuration with the default values.
-   */
-  public JSONParserConfiguration() {
-    super();
-  }
+    /**
+     * Configuration with the default values.
+     */
+    public JSONParserConfiguration() {
+        this(false);
+    }
 
-  @Override
-  protected JSONParserConfiguration clone() {
-    return new JSONParserConfiguration();
-  }
+    /**
+     * Configure the parser with argument overwriteDuplicateKey.
+     *
+     * @param overwriteDuplicateKey Indicate whether to overwrite duplicate key or not.<br>
+     *                              If not, the JSONParser will throw a {@link JSONException}
+     *                              when meeting duplicate keys.
+     */
+    public JSONParserConfiguration(boolean overwriteDuplicateKey) {
+        super();
+        this.overwriteDuplicateKey = overwriteDuplicateKey;
+    }
 
-  @SuppressWarnings("unchecked")
-  @Override
-  public JSONParserConfiguration withMaxNestingDepth(final int maxNestingDepth) {
-    return super.withMaxNestingDepth(maxNestingDepth);
-  }
+    @Override
+    protected JSONParserConfiguration clone() {
+        return new JSONParserConfiguration();
+    }
 
+    @SuppressWarnings("unchecked")
+    @Override
+    public JSONParserConfiguration withMaxNestingDepth(final int maxNestingDepth) {
+        return super.withMaxNestingDepth(maxNestingDepth);
+    }
+
+    public JSONParserConfiguration withOverwriteDuplicateKey(final boolean overwriteDuplicateKey) {
```

Comment:
Please include JavaDocs for this and the next method.
![image](https://github.com/stleary/JSON-java/assets/5107558/42c66bb6-efa0-4ade-9d24-d08ba419b29e)